### PR TITLE
Add rules to find incorrectly formatted units

### PR DESCRIPTION
**Pull request by sebromero**
## What Needs To Be Reviewed
- Please check if the rules detect the wanted patterns.

Some test content could be e.g.

```
400 MHz
400 Mhz
400Mhz

480 *Mbps*
480Mbps

150 kB
150kB
200 KB

12 MB
12mb-test of ram
14 mB

18 μm
test-18μm-test
18μm
```

## How To Give Feedback
Please leave your feedback as a [Github review](https://docs.github.com/en/github/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/about-pull-request-reviews). 
You can add comments to specific lines of content / code and ideally use Github's [suggestion feature](https://docs.github.com/en/github/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/commenting-on-a-pull-request#adding-line-comments-to-a-pull-request). 🙏

### CODE REVIEW

**Comment by sebromero** [comment]
@martab1994 I've added some linter rules in this PR to issue warnings if inconsistent unit formatting is detected.

File: scripts/validation/rules/rules-spelling.yml [hunk 1]
```diff
@@ -5,7 +5,63 @@
...
+  type: warning
+  format: markdown
+  errorMessage: Space between unit and value required
+
+# Mbps / Gbps spelling
+- regex: "(?<!\\S)\\b\\d+\\s?[mg]bps\\b"
+  shouldMatch: false
+  type: warning
+  format: markdown
+  errorMessage: Megabit per second shout be abbreviated mbps, Gigabit per second gbps
```
Comment (jhansson-ard) [inline scripts/validation/rules/rules-spelling.yml:35]:
```suggestion
  errorMessage: 'Megabit per second' should be abbreviated Mbps, 'Gigabit per second' Gbps  
```

File: scripts/validation/rules/rules-spelling.yml [hunk 2]
```diff
@@ -5,7 +5,63 @@
...
+  type: warning
+  format: markdown
+  errorMessage: Space between unit and value required
+
+# MHz / GHz spelling
+- regex: "(?<!\\S)\\b\\d+\\s?[MGmg]hz\\b"
+  shouldMatch: false
+  type: warning
+  format: markdown
+  errorMessage: Megahertz/Gigahertz should be abbreviated as MHz/GHz
```
Comment (jhansson-ard) [inline scripts/validation/rules/rules-spelling.yml:21]:
```suggestion
  errorMessage: 'Megahertz'/'Gigahertz' should be abbreviated as MHz/GHz
```

File: scripts/validation/rules/rules-spelling.yml [hunk 3]
```diff
@@ -5,7 +5,63 @@
...
+  type: warning
+  format: markdown
+  errorMessage: Space between unit and value required
+
+# KB / Kb spelling
+- regex: "(?<!\\S)\\b\\d+\\s?k[bB]\\b"
+  shouldMatch: false
+  type: warning
+  format: markdown
+  errorMessage: Kilobyte should be abbreviated as KB, Kilobit as Kb.
```
Comment (jhansson-ard) [inline scripts/validation/rules/rules-spelling.yml:49]:
```suggestion
  errorMessage: 'Kilobyte' should be abbreviated as KB, 'Kilobit' as Kb.
```

File: scripts/validation/rules/rules-spelling.yml [hunk 4]
```diff
@@ -5,7 +5,63 @@
...
+
+# MB spelling
+- regex: "(?<!\\S)\\b\\d+\\s?m[bB]\\b"
+  shouldMatch: false
+  type: warning
+  format: markdown
+  errorMessage: Megabyte should be abbreviated as MB, Megabit as Mb.
```
Comment (jhansson-ard) [inline scripts/validation/rules/rules-spelling.yml:56]:
```suggestion
  errorMessage: 'Megabyte' should be abbreviated as MB, 'Megabit' as Mb.
```

**Review by jhansson-ard — COMMENTED** [review]
I added some `' '` to better differentiate the 'Megabit per second' compared to the rest of the sentence. Also added this to the other comments. What do you think?

I also updated the Mbps and the Gbps according to our guides.

**Comment by sebromero** [comment]
@jhansson-ard Committed your suggestions. Does it now catch all the desired misspellings?

File: scripts/validation/rules/rules-spelling.yml [hunk 5]
```diff
@@ -5,7 +5,63 @@
   format: markdown
   errorMessage: Megabytes should be either written as 'megabytes' or abbreviated as
```
Comment (jhansson-ard) [inline scripts/validation/rules/rules-spelling.yml:6]:
```suggestion
  errorMessage: 'Megabytes' should be either written as 'megabytes' or abbreviated as MB
```

File: scripts/validation/rules/rules-spelling.yml [hunk 6]
```diff
@@ -5,7 +5,63 @@
...
     'MB'
```
Comment (jhansson-ard) [inline scripts/validation/rules/rules-spelling.yml:7]:
```suggestion

```

**Review by jhansson-ard — COMMENTED** [review]
It seems like this part creates an issue, I updated it here I think

File: scripts/validation/rules/rules-spelling.yml [hunk 7]
```diff
@@ -3,16 +3,85 @@
...
+
+# kHz spelling
+- regex: "(?<!\\S)\\b\\d+\\s?[Kk]hz\\b"
+  shouldMatch: false
+  type: warning
+  format: markdown
+  errorMessage: "'Kilohertz' should be abbreviated as kHz"
+
+# Mbps / Gbps spacing
+- regex: "(?<!\\S)\\b\\d+[MG]bps\\b"
```
Comment (jhansson-ard) [inline scripts/validation/rules/rules-spelling.yml:30]:
```suggestion
- regex: "(?<!\\S)\\b\\d+[MGK]bps\\b"
```

Comment (jhansson-ard) [reply scripts/validation/rules/rules-spelling.yml:30]: @sebromero missed this one!

**Review by jhansson-ard — APPROVED** [review]

**Pull request merged**